model-optimizer/extensions/front/tf/MapFNTransformation.py

```diff
@@ -115,51 +94,45 @@ def transform_keras_rnn_input_slicing(external_match: dict, internal_match: dict
         # remove TensorListFromTensor and pass a tensor to Loop as is
         unstack_node.out_port(0).get_connection().set_source(unstack_node.in_port(0).get_connection().get_source())
 
-    def replace_sub_graph(self, graph: Graph, external_match: dict):
-        loop_node = external_match['while']
-        body_graph = loop_node['body']
-        body_pattern = KerasRNNInputSlicing.get_body_pattern()
-        internal_matches = find_subgraph_match_to_pattern(body_graph, body_pattern)
-
-        # a case of multiple matches is not handled since it is not clear how to select corresponding match
-        if len(internal_matches) == 1:
-            internal_match = internal_matches[0]
-            loop_node = external_match['while']
-            unstack_port_idx = compute_input_port_idx(external_match['unstack'], loop_node)
-            # check that back edges connect correct Parameter and Result nodes in the body
-            # check connections between body input ports and external inputs ports of Loop node
-            if Loop.back_edge_exists(loop_node.back_edges,
-                                     internal_match['increment_iteration_result'].internal_layer_id,
-                                     internal_match['current_iteration'].internal_layer_id) and \
-                    Loop.inter_edge_exists(loop_node.input_port_map, unstack_port_idx,
-                                           internal_match['tensor_list'].internal_layer_id):
-                # only if inter-graph match passed it starts to process the sub-graph
-                KerasRNNInputSlicing.transform_keras_rnn_input_slicing(external_match, internal_match)
-
-
-class KerasRNNOutputConcatenation(FrontReplacementSubgraph):
+    def find_and_replace_pattern(self, graph: Graph):
+        for loop_node in graph.get_op_nodes(op='Loop'):
+            body_graph = loop_node['body']
+            body_pattern = MapFNInputSlicing.get_body_pattern()
+            internal_matches = find_subgraph_match_to_pattern(body_graph, body_pattern)
+
+            for internal_match in internal_matches:
+                # check if TensorListGetItem from the body graph is connected with TensorListFromTensor
+                # from the main graph. If yes, the transformation detects input slicing by this port
+                # and can use Loop axis attribute
+                unstack_node = Loop.get_external_nodes_by_internal_id(loop_node,
+                                                                     internal_match['tensor_list'].internal_layer_id)
+                unstack_node = unstack_node[0] if (len(unstack_node) == 1
+                                                   and unstack_node[0].op == 'TensorListFromTensor') else None
+                if unstack_node is None:
+                    continue
```

Comment:
nit: Let's log some message before `continue` which include node names which we analyze.

Reply:
Fixed. Thanks.